client/src/components/ExpensesTable.jsx

```diff
@@ -92,6 +98,7 @@ const ExpensesForm = () => {
             setOcrData(new Transaction());
         }
         setIsUpdating(false);
+        localStorage.setItem('user', result)
```

Comment:
I don't think this should get set until the response comes back from teh server.